src/sentry/integrations/slack/utils/users.py

```diff
@@ -74,3 +79,63 @@ def get_slack_data_by_user(
                 slack_data_by_user[user] = slack_info_by_email[email]
                 break
     return slack_data_by_user
+
+
+# USING SDK CLIENT
+
+
+@dataclass
+class SlackUserData:
+    email: str
+    team_id: str
+    slack_id: str
+
+
+def format_slack_data_by_user(
+    emails_by_user: Mapping[User, Sequence[str]], slack_info_by_email: dict[str, SlackUserData]
+) -> Mapping[User, SlackUserData]:
+    slack_data_by_user: MutableMapping[User, SlackUserData] = {}
+    for user, emails in emails_by_user.items():
+        # get overlap between user emails and emails in slack
+        user_slack_emails = set(emails) & set(slack_info_by_email.keys())
+        if user_slack_emails:
+            slack_data_by_user[user] = slack_info_by_email[list(user_slack_emails)[0]]
+    return slack_data_by_user
+
+
+def format_slack_info_by_email(users: dict[str, Any]) -> dict[str, SlackUserData]:
+    return {
+        member["profile"]["email"]: SlackUserData(
+            email=member["profile"]["email"], team_id=member["team_id"], slack_id=member["id"]
+        )
+        for member in users
+        if not member["deleted"] and member["profile"].get("email")
+    }
+
+
+def get_slack_data_by_user_via_sdk(
+    integration: Integration | RpcIntegration,
+    organization: Organization | RpcOrganization,
+    emails_by_user: Mapping[User, Sequence[str]],
+) -> Generator[Mapping[User, Mapping[str, str]], None, None] | None:
```

Comment:
The `| None` should be unnecessary unless I'm missing something.

Also, `Generator[Mapping[User, Mapping[str, str]], None, None]` can be simplified to `Iterator[Mapping[User, Mapping[str, str]]]`.